common/src/main/java/com/graphicsfuzz/common/util/FragmentShaderJobToVertexShaderJob.java

```diff
@@ -0,0 +1,226 @@
+import com.graphicsfuzz.common.ast.decl.VariablesDeclaration;
+import com.graphicsfuzz.common.ast.expr.BinOp;
+import com.graphicsfuzz.common.ast.expr.BinaryExpr;
+import com.graphicsfuzz.common.ast.expr.Expr;
+import com.graphicsfuzz.common.ast.expr.FloatConstantExpr;
+import com.graphicsfuzz.common.ast.expr.IntConstantExpr;
+import com.graphicsfuzz.common.ast.expr.ParenExpr;
+import com.graphicsfuzz.common.ast.expr.TypeConstructorExpr;
+import com.graphicsfuzz.common.ast.expr.VariableIdentifierExpr;
+import com.graphicsfuzz.common.ast.stmt.BlockStmt;
+import com.graphicsfuzz.common.ast.stmt.ExprStmt;
+import com.graphicsfuzz.common.ast.type.BasicType;
+import com.graphicsfuzz.common.ast.type.LayoutQualifierSequence;
+import com.graphicsfuzz.common.ast.type.LocationLayoutQualifier;
+import com.graphicsfuzz.common.ast.type.QualifiedType;
+import com.graphicsfuzz.common.ast.type.TypeQualifier;
+import com.graphicsfuzz.common.ast.type.VoidType;
+import com.graphicsfuzz.common.ast.visitors.StandardVisitor;
+import com.graphicsfuzz.common.glslversion.ShadingLanguageVersion;
+import com.graphicsfuzz.common.transformreduce.GlslShaderJob;
+import com.graphicsfuzz.common.transformreduce.ShaderJob;
+import com.graphicsfuzz.common.typing.ScopeTrackingVisitor;
+import com.graphicsfuzz.util.ArgsUtil;
+import com.graphicsfuzz.util.Constants;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import net.sourceforge.argparse4j.ArgumentParsers;
+import net.sourceforge.argparse4j.inf.ArgumentParser;
+import net.sourceforge.argparse4j.inf.ArgumentParserException;
+import net.sourceforge.argparse4j.inf.Namespace;
+
+public final class FragmentShaderJobToVertexShaderJob {
+
+  private FragmentShaderJobToVertexShaderJob() {
+    // Utility class
+  }
+
+  private static Optional<Integer> maybeGetArrayCount(VariableDeclInfo vdi) {
+    if (vdi.hasArrayInfo()) {
+      return Optional.of(vdi.getArrayInfo().getConstantSize());
+    }
+    return Optional.empty();
+  }
+
+  /**
+   * Creates a shader job with the given translation unit as its fragment shader, a pipeline state
+   * that provides a random value for every uniform declared in the shader, and if needed a vertex
+   * shader that provides outputs for the fragment shader's inputs.
+   *
+   * @param tu A fragment shader.
+   * @return A shader job that includes the fragment shader.
+   */
+
+
+  public static ShaderJob convertShaderJob(ShaderJob sj) throws Exception {
+    final PipelineInfo pipelineInfo = sj.getPipelineInfo();
+    final List<TranslationUnit> shaders = new ArrayList<>();
+    
+    // Get the current fragment shader and change it to vertex shader
+    TranslationUnit vertexShader = sj.getFragmentShader().get();
+    vertexShader.setShaderKind(ShaderKind.VERTEX);
+
+    // Create pass-through fragment shader
```

Comment:
I suggest you show the GLSL for the expected fragment shader in a comment; it would be useful as a reference when reading the AST node constructors that follow.

Reply:
Added